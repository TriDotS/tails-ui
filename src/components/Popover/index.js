import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class Popover extends PureComponent {
  static propTypes = {
    trigger: PropTypes.node,
    children: PropTypes.node,
    position: PropTypes.string
  }

  static defaultProps = {
    trigger: 'Click Me',
    position: 'left'
  }

  state = {
    open: false
  }

  toggleOpen = () => {
    this.setState({
      open: !this.state.open
    })
  }

  render() {
    const { open } = this.state
    const { trigger, children, position } = this.props
    return (
      <div
        className="relative inline-block cursor-pointer"
        onClick={this.toggleOpen}
      >
        {trigger}
        {open && (
          <div
            className={`absolute bg-grey-lightest rounded z-10 p-4 ${
              position === 'left' ? 'pin-l top-pin' : 'pin-r top-pin'
            }`}
          >
            {children}
          </div>
        )}
      </div>
    )
  }
}
