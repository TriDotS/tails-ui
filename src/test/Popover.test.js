import Popover from '../components/Popover'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Popover title="A Panel">A popover</Popover>, div)
})

test('popover snapshot', () => {
  const tree = renderer.create(<Popover>A popover</Popover>).toJSON()
  expect(tree).toMatchSnapshot()
})
