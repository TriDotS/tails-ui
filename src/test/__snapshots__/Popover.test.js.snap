// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover snapshot 1`] = `
<div
  className="relative inline-block cursor-pointer"
  onClick={[Function]}
>
  Click Me
</div>
`;
